cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/BuiltInParameterTransformer.java

```diff
@@ -42,6 +42,10 @@ private Object doTransform(String fromValue, Type toValueType, Type originalToVa
             return fromValue;
         }
 
+        if (Character.class.equals(toValueClass) || char.class.equals(toValueClass)) {
+            return fromValue.toCharArray()[0];
+        }
```

Comment:
I think this should throw an exception when the captured string is not a single character.

Otherwise this may lead to surprises when the captured string is not a single character.

This may happen because the regex was capturing more then expected or because the step used a unicode codepoint that requires two bytes to represent.

Reply:
Ideally this should not happen, if it is an empty string it's deciphered type should be string and should not land into character type check construct. But anyways to avoid surprises I've added checks  to transform only when string length is 1 & type is `char`.